QEfficient/peft/auto.py

```diff
@@ -0,0 +1,436 @@
+from QEfficient.utils.cache import QEFF_HOME, to_hashable
+
+logger = logging.getLogger(__name__)
+
+
+class QEffAutoPeftModelForCausalLM(QEFFBaseModel):
+    _pytorch_transforms: List[PytorchTransform] = [CustomOpsTransform, KVCacheTransform, PeftModelInputsTransform]
+    _onnx_transforms: List[OnnxTransform] = [FP16ClipTransform, AdapterWeightsToInputsTransform, SplitTensorsTransform]
+    _hf_auto_class = AutoPeftModelForCausalLM
+
+    @classmethod
+    def transform_names(cls) -> List[str]:
+        # Base class
+        return [x.__name__ for x in cls._pytorch_transforms + cls._onnx_transforms]
+
+    def __init__(self, model: nn.Module):
+        if model.active_peft_config.peft_type != "LORA":
+            raise NotImplementedError("Only LoRA models are supported")
+
+        super().__init__(model)
+
+        self.num_layers = model.config.num_hidden_layers
+        self.adapter_weights = {}
+        self.transform()
+
+    @property
+    def model_name(self) -> str:
+        return self.model.get_base_model().__class__.__name__ + "-lora"
+
+    @property
+    def model_hash(self) -> str:
+        # NOTE: model_config.to_diff_dict() has "_name_or_path" attribute which is the model card name or path.
+        # Using same card name will result in same hash. But, using a relative path for one run and
+        # absolute path for another run will result in different hash.
+        # The added complexity to resolve different paths to same location is not worth pursuing.
+        # Instead, advise the user to always provide same relative paths or absolute paths for local models.
+
+        # Compute the hash with: model_config, peft_config, transforms
+        mhash = hashlib.sha256()
+        mhash.update(to_hashable(self.model.get_base_model().config.to_diff_dict()))
+        mhash.update(to_hashable(self.model.active_peft_config.to_dict()))
+        mhash.update(to_hashable(self.transform_names()))
+        mhash = mhash.hexdigest()[:16]
+        return mhash
+
+    def load_adapter(self, model_id: str, adapter_name: str):
+        self.model.load_adapter(model_id, adapter_name)
+        self.adapter_weights[adapter_name] = {
+            k: v.numpy().astype("float16") for k, v in load_peft_weights(model_id).items()
+        }
+
+    @property
+    def active_adapter(self) -> str:
+        return self.model.active_adapter
+
+    def set_adapter(self, adapter_name: str):
+        self.model.set_adapter(adapter_name)
+
+    def disable_adapter(self):
+        # TODO: Set zero tensors as adapter weights
+        raise NotImplementedError("Disabling adapters not supported currently")
+
+    @classmethod
+    def _from_pretrained(cls, pretrained_name_or_path: str, *args, **kwargs):
+        # Base class
+        model = cls._hf_auto_class.from_pretrained(pretrained_name_or_path, *args, **kwargs)
+        return cls(model)
+
+    @classmethod
+    def from_pretrained(cls, pretrained_name_or_path: str, *args, **kwargs):
```

Comment:
Does not handle Continuous batching case. If we don't want to allow this. raise error in `from_pretrained` when `full_batch_size` is passed

Reply:
I'll raise this error for now.